### PR TITLE
feat(jtk): add sprints remove subcommand

### LINKED ISSUE #320: jtk: add sprints remove subcommand
## Feature Request

`jtk sprints` has `add` but no `remove`. There's no native way to move issues out of a sprint and back to the backlog.

## Current Behavior

The only option is a direct REST call:

```bash
JIRA_URL=$(cat ~/.config/jtk/config.json | jq -r '.url')
JIRA_EMAIL=$(cat ~/.config/jtk/config.json | jq -r '.email')

curl -s -u "$JIRA_EMAIL:$JIRA_API_TOKEN" \
  -X POST "$JIRA_URL/rest/agile/1.0/backlog/issue" \
  -H "Content-Type: application/json" \
  -d '{"issues": ["KEY-123", "KEY-124"]}' | jq
# 204 or empty body = success
```

Note: the Jira model is "move to backlog" rather than "remove from sprint" — the POST target is `/rest/agile/1.0/backlog/issue`, not a sprint endpoint.

## Proposed Solution

```bash
jtk sprints remove KEY-123 KEY-124
```

Mirror the interface of `jtk sprints add` (which moves issues into a sprint). `remove` would POST the issue keys to the backlog endpoint above, returning success or a clear error message.

Acceptance criteria:
- Accepts one or more issue keys as positional args
- Moves them to the backlog (removes from any active sprint)
- Consistent with `jtk sprints add` in argument shape and output style

## API Reference

`POST /rest/agile/1.0/backlog/issue` — [Atlassian docs](https://developer.atlassian.com/cloud/jira/software/rest/api-group-backlog/#api-rest-agile-1-0-backlog-issue-post)

### PR DESCRIPTION
**Pull request by rianjs**
## Summary

- Adds `jtk sprints remove <issue-key>...` — moves issues to the backlog via `POST /rest/agile/1.0/backlog/issue`
- Mirrors `sprints add` in argument shape and output style: post-state display fetches issues after mutation, falls back to confirmation message
- Adds `MoveIssuesToBacklog` API method, `PresentMovedToBacklog` presenter, and comprehensive tests (10 new tests across API and command layers)

## Test plan

- [x] Unit tests pass (`go test ./tools/jtk/...` — 1685 passed, 1 pre-existing failure unrelated)
- [x] Lint passes (`make lint` — 0 issues)
- [x] Build passes (`make build-jtk`)
- [ ] CI green
- [ ] Manual test: `jtk sprints remove <key>` moves issue to backlog

Closes #320

### CODE REVIEW

**Comment by rianjs** [comment]
**Blocker**
None.

**Major**
None.

**Minor**
- [skills/Jira/Workflows/ManageIssueSet.md:122](/Users/rianjs/dev/atlassian-cli/skills/Jira/Workflows/ManageIssueSet.md:122) still says `jtk` has no remove-from-sprint command. That now directly contradicts the new behavior and will mislead agent workflows. Also add `sprints remove` to the sprint command refs in [skills/Jira/CliReference.md:154](/Users/rianjs/dev/atlassian-cli/skills/Jira/CliReference.md:154), [skills/Jira/Workflows/SprintBoard.md:15](/Users/rianjs/dev/atlassian-cli/skills/Jira/Workflows/SprintBoard.md:15), and [README.md:289](/Users/rianjs/dev/atlassian-cli/README.md:289).

**Nit**
- [tools/jtk/api/sprints.go:113](/Users/rianjs/dev/atlassian-cli/tools/jtk/api/sprints.go:113) says “removes from any sprint.” Jira backlog moves remove active/future sprint membership; closed sprint history can remain. Use “current/open sprint” or “active/future sprint membership” to avoid overpromising.

**Comment by rianjs** [comment]
## TDD Assessment

**Covered**
- `MoveIssuesToBacklog` API method: happy path (correct URL, body shape) and error path (500 → wrapped error message)
- `newRemoveCmd`: cobra metadata (Use, Short) verified
- `runRemove`: multi-issue success, single-issue success, `--id-only` mode, API error propagation, and the fallback path (GetIssue failure after backlog move)
- `PresentMovedToBacklog`: singular vs. plural message branching covered indirectly through integration tests

**Gaps**
- `MoveIssuesToBacklog` with an **empty slice** (`[]string{}`) — the API call is still made; behavior is untested
- Context cancellation mid-backoff in `runRemove` is not directly exercised (the `ctx.Done()` branch inside the retry loop)
- `PresentMovedToBacklog` has no dedicated unit test; singular/plural logic is only exercised via the higher-level `runRemove` tests
- No test for `newRemoveCmd` enforcing `cobra.MinimumNArgs(1)` (zero-arg invocation should error at the cobra level)

**Verdict**
👍 Coverage is solid for the core happy and error paths; the gaps are minor edge cases (empty input, arg validation, context cancel) that carry low production risk but would complete the suite.

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 1]
```diff
@@ -1303,6 +1303,226 @@ func TestCurrentCmd_BoardEnrichmentCobraLevel(t *testing.T) {
...
+func TestNewRemoveCmd(t *testing.T) {
+	opts := &root.Options{}
+	cmd := newRemoveCmd(opts)
+
+	testutil.Equal(t, cmd.Use, "remove <issue-key>...")
+	testutil.NotEmpty(t, cmd.Short)
+}
+
+func TestRunRemove_Success(t *testing.T) {
+	postDone := false
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go]: 🟡 **Medium** (harness-engineering:harness-architecture-reviewer): Data race on `postDone` bool in TestRunRemove_Success (line 1317) and TestRunRemove_SingleIssue (line 1364): httptest.Server dispatches each request in its own goroutine, so the write (`postDone = true`) in the POST handler and the read in subsequent GET handlers occur in separate goroutines without synchronization. This will be flagged by `go test -race`. Use `atomic.Bool` or a `sync.Mutex`.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 2]
```diff
@@ -1303,6 +1303,226 @@ func TestCurrentCmd_BoardEnrichmentCobraLevel(t *testing.T) {
...
+	var stdout bytes.Buffer
+	opts := &root.Options{Output: "table", Stdout: &stdout, Stderr: &bytes.Buffer{}}
+	opts.SetAPIClient(client)
+
+	err = runRemove(context.Background(), opts, client, []string{"PROJ-101"})
+	testutil.NotNil(t, err)
+	testutil.Contains(t, err.Error(), "moving issues to backlog")
+}
+
+func TestRunRemove_FetchFails_Fallback(t *testing.T) {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:1441]: 🟡 **Medium** (harness-engineering:harness-enforcement-reviewer): This test file is missing `func init() { mutation.BackoffSchedule = []time.Duration{0, 0, 0, 0} }`, unlike every other test file that exercises the backoff loop (assign_test.go, transitions_test.go, mutation_test.go). Tests that exhaust the schedule (FetchFails_Fallback, CobraEntryPoint, AgentOutput) will sleep through real delays (700ms+ each), unnecessarily slowing the suite. This also explains why some tests omit `t.Parallel()` — but the root fix is to zero the schedule.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 3]
```diff
@@ -1303,6 +1303,226 @@ func TestCurrentCmd_BoardEnrichmentCobraLevel(t *testing.T) {
...
+	requestCount := 0
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): `requestCount` is declared and incremented in TestRunRemove_FetchFails_Fallback but never asserted on. Either assert the expected retry count to verify the backoff loop ran the expected number of times, or remove the dead variable.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/sprints/sprints_test.go [hunk 4]
```diff
@@ -1303,6 +1303,226 @@ func TestCurrentCmd_BoardEnrichmentCobraLevel(t *testing.T) {
...
+
+	err = runRemove(context.Background(), opts, client, []string{"PROJ-101"})
+	testutil.RequireNoError(t, err)
+
+	testutil.Contains(t, stdout.String(), "PROJ-101")
+}
+
+func TestRunRemove_IDOnly(t *testing.T) {
+	t.Parallel()
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/sprints/sprints_test.go:1403]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): TestRunRemove_Success, TestRunRemove_SingleIssue, TestRunRemove_FetchFails_Fallback, and TestRemoveCmd_CobraEntryPoint are missing `t.Parallel()`, inconsistent with other tests in this file. Once the `postDone` race is fixed (via atomic) and BackoffSchedule is zeroed, all of these can safely call `t.Parallel()`.

*Reply to this thread when addressed.*

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `81f5e6a`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 2 |
| harness-engineering:harness-enforcement-reviewer | 2 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (2 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1317`
> Data race on `postDone` bool in TestRunRemove_Success (line 1317) and TestRunRemove_SingleIssue (line 1364): httptest.Server dispatches each request in its own goroutine, so the write (`postDone = true`) in the POST handler and the read in subsequent GET handlers occur in separate goroutines without synchronization. This will be flagged by `go test -race`. Use `atomic.Bool` or a `sync.Mutex`.

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1397`
> TestRunRemove_Success, TestRunRemove_SingleIssue, TestRunRemove_FetchFails_Fallback, and TestRemoveCmd_CobraEntryPoint are missing `t.Parallel()`, inconsistent with other tests in this file. Once the `postDone` race is fixed (via atomic) and BackoffSchedule is zeroed, all of these can safely call `t.Parallel()`.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (2 findings)</strong></summary>

**⚠️ Should Fix** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1435`
> This test file is missing `func init() { mutation.BackoffSchedule = []time.Duration{0, 0, 0, 0} }`, unlike every other test file that exercises the backoff loop (assign_test.go, transitions_test.go, mutation_test.go). Tests that exhaust the schedule (FetchFails_Fallback, CobraEntryPoint, AgentOutput) will sleep through real delays (700ms+ each), unnecessarily slowing the suite. This also explains why some tests omit `t.Parallel()` — but the root fix is to zero the schedule.

**💡 Suggestion** - `tools/jtk/internal/cmd/sprints/sprints_test.go:1436`
> `requestCount` is declared and incremented in TestRunRemove_FetchFails_Fallback but never asserted on. Either assert the expected retry count to verify the backoff loop ran the expected number of times, or remove the dead variable.

</details>

*2 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 50s | $1.10</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 50s wall · 7m 52s compute (Reviewers: 1m 50s · Synthesis: 57s) |
| Cost | $1.10 |
| Tokens | 408.4k in / 28.9k out |
| Turns | 25 |

</details>

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go]: Fixed: replaced `postDone bool` with `atomic.Bool` in both TestRunRemove_Success and TestRunRemove_SingleIssue.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1441]: Fixed: added `func init() { mutation.BackoffSchedule = []time.Duration{0, 0, 0, 0} }` to zero all delays.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go]: Fixed: removed the dead `requestCount` variable.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1403]: Fixed: added `t.Parallel()` to all new remove tests.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 1 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go]: **Settled**: monit-reviewer flagged a data race on `postDone` bool in TestRunRemove_Success and TestRunRemove_SingleIssue. rianjs fixed it by replacing `postDone bool` with `atomic.Bool` in both tests.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 2 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1441]: **Settled**: monit-reviewer flagged that the test file was missing the `init()` function to zero `mutation.BackoffSchedule`, causing unnecessary real delays. rianjs fixed it by adding the init function.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 3 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go]: **Settled**: monit-reviewer flagged a dead `requestCount` variable that was incremented but never asserted on. rianjs fixed it by removing the variable.

File: tools/jtk/internal/cmd/sprints/sprints_test.go (hunk 4 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/sprints/sprints_test.go:1403]: **Settled**: monit-reviewer flagged that the new remove tests were missing `t.Parallel()`. rianjs fixed it by adding `t.Parallel()` to all new remove tests.

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `c28d029` | **Previous:** `81f5e6a` (incremental)

### Summary

No issues found.

*6 PR discussion threads considered.*

---
<details>
<summary>Completed in 3m 31s | $1.04</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 3m 31s wall · 7m 35s compute (Reviewers: 2m 09s · Synthesis: 1m 05s) |
| Cost | $1.04 |
| Tokens | 339.4k in / 25.5k out |
| Turns | 19 |

</details>

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `306d6ae` | **Previous:** `c28d029` (incremental)

### Summary

No issues found.

*6 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 47s | $1.06</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 3 · Session resumed |
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 47s wall · 5m 58s compute (Reviewers: 1m 52s · Synthesis: 52s) |
| Cost | $1.06 |
| Tokens | 564.8k in / 20.7k out |
| Turns | 28 |

</details>
